optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/OptaShiftRosteringEntryPoint.java

```diff
@@ -23,19 +25,24 @@
     }
 
     @Inject
-    private Document document;
+    private NavigationController navigationController;
+
+    @Inject
+    private Event<PageChange> pageChangeEvent;
 
     @Inject
-    private MenuPanel menuPanel;
+    private TenantStore tenantStore;
 
     @PostConstruct
     public void onModuleLoad() {
-        document.getBody().appendChild(menuPanel.getElement());
-        /*GWT.setUncaughtExceptionHandler(new
-                GWT.UncaughtExceptionHandler() {
-                public void onUncaughtException(Throwable e) {
-                  ErrorPopup.show(Arrays.asList(e.getStackTrace()).stream().reduce("", (a,b) -> a.toString() + "\n" + b.toString(), (a,b) -> a + "\n" + b));
-              }
-          });*/
+        tenantStore.init(); //FIXME: Shouldn't this call be made by the Container once it's annotated with @PostConstruct?
```

Comment:
I think the scopes of this and the `TenantStore` are inverted. It looks like the the `TenantStore` should be an `@EntryPoint` with a `@PostConstruct` that initializes itself, and this class should be an `@ApplicationScoped` bean that will be created by the container when the `TenantsReady` event fires.

Reply:
That makes sense. In my reasoning, when an `@EntryPoint` bean depends on a `@ApplicationScoped` bean, the `@PostConstruct` method of the `@ApplicationScoped` bean is invoked before the `@PostConstruct` of the `@EntryPoint` bean. I guess that makes sense too?

Reply:
You need to be careful calling another bean's `@PostConstruct`. There is no special logic to ensure it's only called once, so you should only do this if your sure the method is idempotent.

Reply:
That's valuable information. Thanks for sharing it :)